src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -690,7 +699,7 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
         {
             DeconstructionVariable locals = BindDeconstructionVariables(left, diagnostics, ref declaration, ref expression);
```

Comment:
I think we need NestedVariables because the variables need to be updated (by the inference process). NestedVariables uses a builder, but tuple expression is immutable.